### PR TITLE
New SASS to CSS IA

### PR DESCRIPTION
**Pull request by akanshgulati**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [x] New Instant Answer
    - [ ] Other: **`New SASS to CSS Instant Answer`**
###### Description of changes
This is the new IA for converting SASS to CSS

What changes does this PR introduce? 

###### People to notify

@moollaza  @hchienjo 

---
Instant Answer Page: https://duck.co/ia/view/sass_to_css
Maintainer: @moollaza

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
@akanshgulati which of the files need to be loaded for all this to work? I assume we can get rid of the un-minified files?

I will have to add them internally so that they can be loaded via `DDG.require( 'libraryName', function { ... })` will load them.

**Comment by akanshgulati** [comment]
https://cdnjs.cloudflare.com/ajax/libs/sass.js/0.9.12/sass.min.js
https://cdnjs.cloudflare.com/ajax/libs/sass.js/0.9.12/sass.worker.min.js

These are the two files which need to be added in same folder.

File: t/SassToCss.t [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+sub build_structured_answer {
+    my @test_params = @_;
+
+    return "",
+        structured_answer => {
+
+            id => "sass_to_css",
+
+            data => {
+                goodie_version => 999,
```
Comment (moollaza) [inline t/SassToCss.t]: @akanshgulati please remove this so the tests are passing again

File: share/goodie/sass_to_css/sass.min.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/*! sass.js - v0.9.12 (e1a0112) - built 2016-08-14
```
Comment (moollaza) [inline share/goodie/sass_to_css/sass.min.js]: Please remove this and the other libraries from the PR as they're no longer needed

File: lib/DDG/Goodie/SassToCss.pm [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+use warnings;
+
+zci answer_type => 'sass_to_css';
+
+zci is_cached => 1;
+
+
+
+# Triggers - http://docs.duckduckhack.com/walkthroughs/calculation.html#triggers
+triggers any => share('triggers.txt')->slurp;
```
Comment (moollaza) [inline lib/DDG/Goodie/SassToCss.pm]: I think we should use `startend` here -- do you have any example queries where the trigger phrase is in the middle of a larger query?

File: lib/DDG/Goodie/SassToCss.pm [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+handle remainder => sub {
+
+    # Return unless the remainder is empty or contains online or tool
+    return unless ( $_ =~ /(^$|online|tool)/i );
+
+    return '',
+        structured_answer => {
+
+            id => "sass_to_css",
```
Comment (moollaza) [inline lib/DDG/Goodie/SassToCss.pm]: Goodies don't need to specify the `ID` -- it comes from the IA Page 👍

File: share/goodie/sass_to_css/sass_to_css.js [hunk 5]
```diff
@@ -28,7 +27,6 @@ DDH.sass_to_css.build = function(ops) {
             // Load library when the IA is shown for the first time
 
             DDG.require('sass.js', function(){
-                sass = new window.Sass();
```
Comment (moollaza) [inline share/goodie/sass_to_css/sass_to_css.js]: @akanshgulati you still need this, or you'll need to move the `$validateButton.click` function definition inside the `DDG.require()` call.

the `Sass` variable is only guaranteed to be defined inside there, as that callback function executes after sass.js has loaded.

On a slow connection for example, `DDG.require` could take a while to load sass.js, but the click function will still be defined and `Sass` isn't guaranteed to exist yet.

Comment (akanshgulati) [inline share/goodie/sass_to_css/sass_to_css.js]: @moollaza you are right on this, I even thought on this and making changes to add click event only when sass.js is loaded.
Moreover, loading sass.js initially is also not efficient, I am changing the library to load when input has finally some content.

**Comment by moollaza** [comment]
@akanshgulati this is now live on Beta 👍 

I noticed that it only seems to accept SCSS and not SASS input? We should ensure it supports both syntaxes.

I borrowed the snippet of code from the [SASS Lang website](http://sass-lang.com/guide)

sass_to_css_at_duckduckgo

**Comment by daxtheduck** [comment]
@akanshgulati this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Pull request merged**